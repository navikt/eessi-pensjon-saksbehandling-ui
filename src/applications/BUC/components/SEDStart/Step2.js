import React, { useState } from 'react'
import { Flatknapp, Hovedknapp, Row } from 'components/ui/Nav'

const Step2 = (props) => {

  const { actions, t, loading, backStep } = props

  const onBackButtonClick = () => {
    backStep()
  }

  const onForwardButtonClick = () => {
    console.log('Forward')
  }

  const onCancelButtonClick = () => {
    actions.resetBuc()
    actions.setMode('list')
  }

  const allowedToForward = () => {
    return true
  }

  return <Row>
    <div className='col-md-12'>
      <Hovedknapp
        id='a-buc-c-sedstart__forward-button-id'
        className='a-buc-c-sedstart__forward-button'
        disabled={!allowedToForward()}
        spinner={loading.creatingSed}
        onClick={ onForwardButtonClick }>
        {t('buc:form-orderSED')}
      </Hovedknapp>
      <Flatknapp
        id='a-buc-c-sedstart__back-button-id'
        className='a-buc-c-sedstart__back-button'
        onClick={onBackButtonClick}>{t('ui:back')}
      </Flatknapp>
      <Flatknapp
        id='a-buc-c-sedstart__cancel-button-id'
        className='a-buc-c-sedstart__cancel-button'
        onClick={onCancelButtonClick}>{t('ui:cancel')}
      </Flatknapp>
    </div>
 </Row>

}

export default Step2
