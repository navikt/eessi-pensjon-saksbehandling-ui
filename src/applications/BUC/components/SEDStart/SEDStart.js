import React, { useState, useEffect } from 'react'
import PT from 'prop-types'
import { connect, bindActionCreators } from 'store'
import _ from 'lodash'
import Step1 from './Step1'
import Step2 from './Step2'
import { Nav } from 'eessi-pensjon-ui'
import * as bucActions from 'actions/buc'
import * as uiActions from 'actions/ui'
import * as storageActions from 'actions/storage'
import SEDAttachmentSender from 'applications/BUC/components/SEDAttachmentSender/SEDAttachmentSender'
import PInfoUtil from 'applications/BUC/components/SEDP4000/P4000Payload'
import { IS_TEST } from 'constants/environment'
import * as storage from 'constants/storage'

const mapStateToProps = /* istanbul ignore next */ (state) => {
  return {
    attachments: state.buc.attachments,
    bucsInfoList: state.buc.bucsInfoList,
    countryList: state.buc.countryList,
    institutionList: state.buc.institutionList,
    attachmentsError: state.buc.attachmentsError,
    loading: state.loading,
    locale: state.ui.locale,
    sed: state.buc.sed,
    sedList: state.buc.sedList,
    p4000info: state.buc.p4000info,
    avdodfnr: state.app.params.avdodfnr
  }
}

const mapDispatchToProps = /* istanbul ignore next */ (dispatch) => {
  return { actions: bindActionCreators({ ...storageActions, ...bucActions, ...uiActions }, dispatch) }
}

export const SEDStart = (props) => {
  const { actions, aktoerId, avdodfnr, attachments, attachmentsError, bucs, bucsInfoList, countryList, currentBuc, currentSed } = props
  const { initialAttachments = {}, initialSed = undefined, initialStep = 0, institutionList, institutionNames } = props
  const { loading, p4000info, sakId, sed, setMode, t, vedtakId } = props

  const [_sed, setSed] = useState(initialSed)
  const [_institutions, setInstitutions] = useState(
    bucs[currentBuc] && bucs[currentBuc].institusjon
      ? bucs[currentBuc].institusjon
        .filter(inst => inst.institution !== bucs[currentBuc].creator.institution)
        .map(inst => inst.institution) : [])
  const [_countries, setCountries] = useState(
    bucs[currentBuc] && bucs[currentBuc].institusjon
      ? _.uniq(bucs[currentBuc].institusjon.map(inst => inst.country)) : [])
  const [_vedtakId, setVedtakId] = useState(parseInt(vedtakId, 10))
  const [_attachments, setAttachments] = useState(initialAttachments)
  const [step, setStep] = useState(initialStep)
  const [validation, setValidation] = useState({})

  const [showButtons, setShowButtons] = useState(true)
  const [sedSent, setSedSent] = useState(false)
  const [sendingAttachments, setSendingAttachments] = useState(false)
  const [attachmentsSent, setAttachmentsSent] = useState(false)
  const [mounted, setMounted] = useState(false)

  const buc = _.cloneDeep(bucs[currentBuc])

  useEffect(() => {
    if (_.isEmpty(countryList) && !loading.gettingCountryList) {
      actions.getCountryList(buc.type)
    }
  }, [actions, countryList, loading])

  useEffect(() => {
    if (!mounted) {
      if (!currentSed || !currentBuc) {
        actions.getSedList(buc)
      } else {
        actions.setSedList(
          bucs[currentBuc].seds
            .filter(sed => sed.parentDocumentId === currentSed)
            .map(sed => sed.type)
        )
      }
      setMounted(true)
    }
  }, [mounted, actions, buc, bucs, currentBuc, currentSed])

  useEffect(() => {
    // mark sed as sent
    if (sed && !sedSent) {
      setSedSent(true)
    }
  }, [sed, sedSent])

  useEffect(() => {
    // if sed is sent, we can start sending attachments
    if (sedSent && !attachmentsSent) {
      // no attachments to send - conclude
      if (_.isEmpty(_attachments) || !_attachments.joark || _.isEmpty(_attachments.joark)) {
        if (!IS_TEST) {
          console.log('SEDStart: No attachments to send, concluding')
        }
        setAttachmentsSent(true)
        return
      }
      // mark state as sending attachments
      setSendingAttachments(true)
      if (!IS_TEST) {
        console.log('SEDStart: Marking setSendingAttachments as true')
      }
    }
  }, [_attachments, attachmentsSent, sedSent])

  useEffect(() => {
    // cleanup after attachments sent
    if (sedSent && attachmentsSent) {
      if (!IS_TEST) {
        console.log('SEDStart: Attachments sent, cleaning up')
      }
      actions.resetSed()
      actions.resetSedAttachments()
      actions.fetchBucs(aktoerId)
      if (avdodfnr) {
        actions.fetchBucs(avdodfnr)
      }
      if (!_.isEmpty(bucsInfoList) &&
        bucsInfoList.indexOf(aktoerId + '___' + storage.NAMESPACE_BUC + '___' + storage.FILE_BUCINFO) >= 0) {
        actions.fetchBucsInfo(aktoerId, storage.NAMESPACE_BUC, storage.FILE_BUCINFO)
      }
      setMode('bucedit')
    }
  }, [actions, aktoerId, attachmentsSent, avdodfnr, bucsInfoList, sedSent, setMode])

  if (_.isEmpty(bucs) || !currentBuc) {
    return null
  }

  const bucHasSedsWithAtLeastOneInstitution = () => {
    if (buc.seds) {
      return _(buc.seds).find(sed => {
        return _.isArray(sed.participants) && !_.isEmpty(sed.participants)
      }) !== undefined
    }
    return false
  }

  const sedNeedsVedtakId = () => {
    return _sed === 'P6000' || _sed === 'P7000'
  }

  const sedCanHaveAttachments = () => {
    const sedsWithoutAttachments = ['P4000', 'P5000', 'P7000', 'H070']
    return _sed !== undefined && !_.includes(sedsWithoutAttachments, _sed)
  }

  const convertInstitutionIDsToInstitutionObjects = () => {
    const institutions = []
    _institutions.forEach(item => {
      Object.keys(institutionList).forEach(landkode => {
        const found = _.find(institutionList[landkode], { id: item })
        if (found) {
          institutions.push({
            country: found.landkode,
            institution: found.id,
            name: found.navn
          })
        }
      })
    })
    return institutions
  }

  const onForwardButtonClick = () => {
    if (_.isEmpty(validation)) {
      const institutions = convertInstitutionIDsToInstitutionObjects()
      const payload = {
        sakId: sakId,
        buc: buc.type,
        sed: _sed,
        institutions: institutions,
        aktoerId: aktoerId,
        euxCaseId: buc.caseId
      }

      if (_sed === 'P4000' && p4000info) {
        const periods = new PInfoUtil(p4000info.stayAbroad, t).generatePayload()
        payload.periodeInfo = periods.periodeInfo
      }
      if (sedNeedsVedtakId()) {
        payload.vedtakId = _vedtakId
      }
      if (avdodfnr) {
        payload.avdodfnr = avdodfnr
      }
      if (currentSed) {
        actions.createReplySed(payload, currentSed)
      } else {
        actions.createSed(payload)
      }
    }
  }

  const onNextButtonClick = () => {
    setStep(step + 1)
  }

  const onBackButtonClick = () => {
    setStep(step - 1)
  }

  const onCancelButtonClick = () => {
    actions.resetSed()
    setMode('bucedit')
  }

  const createSedNeedsMoreSteps = () => {
    return false // _sed === 'P4000'
  }

  const allowedToForward = () => {
    if (step === 0) {
      return _sed && _.isEmpty(validation) &&
       (bucHasSedsWithAtLeastOneInstitution() || !_.isEmpty(_institutions)) &&
       !loading.creatingSed && !sendingAttachments &&
       (sedNeedsVedtakId() ? _.isNumber(_vedtakId) && !_.isNaN(_vedtakId) : true)
    }
    return false
  }

  return (
    <Nav.Row className='a-buc-c-sedstart'>
      {step === 0 ? (
        <Step1
          {...props}
          _sed={_sed} setSed={setSed} currentSed={currentSed} buc={buc}
          _countries={_countries} setCountries={setCountries} countryList={countryList}
          _institutions={_institutions} setInstitutions={setInstitutions} institutionList={institutionList}
          institutionNames={institutionNames}
          _attachments={_attachments} setAttachments={setAttachments}
          validation={validation} setValidation={setValidation}
          sedNeedsVedtakId={sedNeedsVedtakId}
          sedCanHaveAttachments={sedCanHaveAttachments}
          vedtakId={_vedtakId} setVedtakId={setVedtakId}
        />
      ) : null}
      {step === 1 ? (
        <Step2
          {...props}
          _sed={_sed}
          buc={buc}
          showButtons={showButtons} setShowButtons={setShowButtons}
          validation={validation} setValidation={setValidation}
        />
      ) : null}
      {sendingAttachments || attachmentsSent ? (
        <SEDAttachmentSender
          className='ml-3 w-50'
          attachmentsError={attachmentsError}
          sendAttachmentToSed={actions.sendAttachmentToSed}
          payload={{
            aktoerId: aktoerId,
            rinaId: buc.caseId,
            rinaDokumentId: sed.id
          }}
          allAttachments={_attachments.joark}
          savedAttachments={attachments}
          onFinished={() => setAttachmentsSent(true)}
          t={t}
        />
      ) : null}
      {showButtons ? (
        <div className='col-md-12 mt-4'>
          <Nav.Hovedknapp
            id='a-buc-c-sedstart__forward-button-id'
            className='a-buc-c-sedstart__forward-button'
            disabled={!allowedToForward()}
            spinner={loading.creatingSed || sendingAttachments}
            onClick={createSedNeedsMoreSteps() ? onNextButtonClick : onForwardButtonClick}
          >
            {loading.creatingSed ? t('buc:loading-creatingSED')
              : sendingAttachments ? t('buc:loading-sendingSEDattachments')
                : createSedNeedsMoreSteps() ? t('ui:next')
                  : t('buc:form-orderSED')}
          </Nav.Hovedknapp>
          {step > 0 ? (
            <Nav.Flatknapp
              id='a-buc-c-sedstart__back-button-id'
              className='a-buc-c-sedstart__back-button'
              onClick={onBackButtonClick}
            >{t('ui:back')}
            </Nav.Flatknapp>
          ) : null}
          <Nav.Flatknapp
            id='a-buc-c-sedstart__cancel-button-id'
            className='a-buc-c-sedstart__cancel-button'
            onClick={onCancelButtonClick}
          >{t('ui:cancel')}
          </Nav.Flatknapp>
        </div>
      ) : null}
    </Nav.Row>
  )
}

SEDStart.propTypes = {
  actions: PT.object.isRequired,
  aktoerId: PT.string.isRequired,
  avdodfnr: PT.string,
  attachments: PT.array,
  bucs: PT.object.isRequired,
  bucsInfoList: PT.array,
  countryList: PT.array,
  currentBuc: PT.string.isRequired,
  initialAttachments: PT.object,
  institutionList: PT.object,
  institutionNames: PT.object,
  loading: PT.object.isRequired,
  p4000info: PT.object,
  sakId: PT.string.isRequired,
  sed: PT.object,
  t: PT.func.isRequired,
  vedtakId: PT.string
}

export default connect(mapStateToProps, mapDispatchToProps)(SEDStart)
