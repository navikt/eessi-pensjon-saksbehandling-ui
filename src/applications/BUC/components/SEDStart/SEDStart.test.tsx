import { createSed, getCountryList, getSedList, resetSed, sendAttachmentToSed } from 'actions/buc'
import { Bucs } from 'declarations/buc'
import { JoarkFile } from 'declarations/joark'
import { mount, ReactWrapper } from 'enzyme'
import _ from 'lodash'
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import sampleBucs from 'resources/tests/sampleBucs'
import sampleP4000info from 'resources/tests/sampleP4000info'
import targetP4000info from 'resources/tests/targetP4000info'
import { SEDStart, SEDStartProps, SEDStartSelector } from './SEDStart'

jest.mock('applications/BUC/components/SEDP4000/SEDP4000', () => ({ children }: {children: JSX.Element}) => (
  <div className='mock-sedP4000'>
    {children}
  </div>
))

jest.mock('actions/buc', () => ({
  createSed: jest.fn(),
  getCountryList: jest.fn(),
  getSedList: jest.fn(),
  resetSed: jest.fn(),
  sendAttachmentToSed: jest.fn(),
  getInstitutionsListForBucAndCountry: jest.fn()
}))

jest.mock('react-redux');
(useDispatch as jest.Mock).mockImplementation(() => jest.fn())

const defaultSelector: SEDStartSelector = {
  attachments: {sed: [], joark :[]},
  attachmentsError: false,
  avdodfnr: undefined,
  bucsInfoList: undefined,
  countryList: [],
  institutionList: {},
  loading: {},
  locale: 'nb',
  sakId: '123',
  sed: undefined,
  sedList: undefined,
  p4000info: sampleP4000info,
  vedtakId: undefined
};

(useSelector as jest.Mock).mockImplementation(() => (defaultSelector))

describe('applications/BUC/components/SEDStart/SEDStart', () => {
  let wrapper: ReactWrapper
  const mockBucs: Bucs = _.keyBy(sampleBucs, 'caseId')

  const initialMockProps: SEDStartProps = {
    aktoerId: '123',
    bucs: mockBucs,
    currentBuc: '195440',
    initialAttachments: {
      joark: [{
        tittel: 'tittel',
        tema: 'tema',
        datoOpprettet: new Date(2020, 1, 1),
        journalpostId: '456',
        dokumentInfoId: '789',
        variant: {
          variantformat: 'ARKIV',
          filnavn: 'mockFilename'
        }
      }]
    },
    initialSed: 'P2000',

    setMode: jest.fn()
  }

  beforeEach(() => {
    wrapper = mount(<SEDStart {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('UseEffect: getCountryList', () => {
    expect(getCountryList).toHaveBeenCalled()
  })

  it('UseEffect: getSedList', () => {
    expect(getSedList).toHaveBeenCalled()
  })

  it('UseEffect: sendAttachmentToSed', () => {
    expect(sendAttachmentToSed).not.toHaveBeenCalled();
    (useSelector as jest.Mock).mockImplementation(() => ({
      ...defaultSelector,
      sed: {
        id: 'mockSedId'
      }
    }))
    wrapper = mount(<SEDStart {...initialMockProps} />)
    expect(sendAttachmentToSed).toHaveBeenCalledWith({
      aktoerId: initialMockProps.aktoerId,
      rinaId: initialMockProps.currentBuc,
      rinaDokumentId: 'mockSedId',
      journalpostId: (initialMockProps.initialAttachments!.joark[0] as JoarkFile).journalpostId,
      dokumentInfoId: (initialMockProps.initialAttachments!.joark[0] as JoarkFile).dokumentInfoId,
      variantformat: (initialMockProps.initialAttachments!.joark[0] as JoarkFile).variant.variantformat
    }, initialMockProps.initialAttachments!.joark[0])
  })

  it('With a BUC with SEDs that have NO participants, demand a institution', () => {
    (useSelector as jest.Mock).mockImplementation(() => (defaultSelector))
    const mockBucsWithNoParticipants: Bucs = _.cloneDeep(mockBucs)
    mockBucsWithNoParticipants[initialMockProps.currentBuc!].institusjon = []
    mockBucsWithNoParticipants[initialMockProps.currentBuc!].seds = []
    wrapper = mount(<SEDStart {...initialMockProps} bucs={mockBucsWithNoParticipants} />)
    expect(wrapper.find('#a-buc-c-sedstart__forward-button-id').hostNodes().props().disabled).toEqual(true)
    wrapper.find('#a-buc-c-sedstart__sed-select-id').hostNodes().simulate('change', { target: { value: 'mockSed' } })
    expect(wrapper.find('#a-buc-c-sedstart__forward-button-id').hostNodes().props().disabled).toEqual(true)
  })

  it('With a BUC with SEDs that have participants, no need to demand a institution', () => {
    wrapper = mount(<SEDStart {...initialMockProps} initialSed={undefined} />)
    expect(wrapper.find('#a-buc-c-sedstart__forward-button-id').hostNodes().props().disabled).toEqual(true)
    wrapper.find('#a-buc-c-sedstart__sed-select-id').hostNodes().simulate('change', { target: { value: 'mockSed' } })
    expect(wrapper.find('#a-buc-c-sedstart__forward-button-id').hostNodes().props().disabled).toEqual(false)
  })

  it('With SED P4000 we do not need more steps', () => {
    expect(wrapper.find('#a-buc-c-sedstart__forward-button-id').hostNodes().render().text()).toEqual('buc:form-orderSED')
    wrapper.find('#a-buc-c-sedstart__sed-select-id').hostNodes().simulate('change', { target: { value: 'P4000' } })
    expect(wrapper.find('#a-buc-c-sedstart__forward-button-id').hostNodes().render().text()).toEqual('buc:form-orderSED')
  })

  it('With SED P4000 we get a proper submit', () => {
    (createSed as jest.Mock).mockReset()
    wrapper.find('#a-buc-c-sedstart__sed-select-id').hostNodes().simulate('change', { target: { value: 'P4000' } })
    wrapper.find('#a-buc-c-sedstart__forward-button-id').hostNodes().simulate('click')
    expect(createSed).toHaveBeenCalledWith({
      aktoerId: '123',
      buc: 'P_BUC_02',
      euxCaseId: '195440',
      institutions: [],
      periodeInfo: targetP4000info.trygdetid,
      sed: 'P4000',
      sakId: '123'
    })
  })

  it('Has proper HTML structure', () => {
    expect(wrapper.exists('.a-buc-c-sedstart')).toBeTruthy()
    expect(wrapper.exists('#a-buc-c-sedstart__forward-button-id')).toBeTruthy()
    expect(wrapper.exists('#a-buc-c-sedstart__cancel-button-id')).toBeTruthy()
  })

  it('Renders with step 1', () => {
    expect(wrapper.find('Step1')).toBeTruthy()
  })

  it('Renders with step 2', () => {
    wrapper = mount(<SEDStart {...initialMockProps} initialStep={1} />)
    expect(wrapper.find('Step2')).toBeTruthy()
  })

  it('Creates sed when forward button is clicked ', () => {
    expect(wrapper.find('#a-buc-c-sedstart__forward-button-id').hostNodes().prop('disabled')).toBeFalsy()
    wrapper.find('#a-buc-c-sedstart__forward-button-id').hostNodes().simulate('click')
    expect(createSed).toHaveBeenCalledWith({
      aktoerId: '123',
      buc: 'P_BUC_02',
      euxCaseId: '195440',
      institutions: [],
      sakId: '123',
      sed: 'P2000'
    })
  })

  it('Cancels sed when cancel button is clicked ', () => {
    wrapper.find('#a-buc-c-sedstart__cancel-button-id').hostNodes().simulate('click')
    expect(resetSed).toHaveBeenCalled()
    expect(initialMockProps.setMode).toHaveBeenCalledWith('bucedit')
  })
})
