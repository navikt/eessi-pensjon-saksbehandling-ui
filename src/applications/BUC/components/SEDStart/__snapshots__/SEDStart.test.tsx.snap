// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/SEDStart/SEDStart Render: match snapshot 1`] = `
.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g7[id="c17"] {
  content: "c18,";
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -0.5rem;
  margin-right: -0.5rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g21[id="c4"] {
  content: "c5,";
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

ui-fss.g25[id="c6"] {
  content: "c7,";
}

.c14 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g31[id="c13"] {
  content: "c14,";
}

.c9 {
  margin-bottom: 2rem;
}

.c10 {
  margin-bottom: 1rem;
}

.c19 {
  margin-bottom: 1.5rem;
}

ui-fss.g32[id="c8"] {
  content: "c9,c10,c19,";
}

.c16 .skjemaelement--feilmelding .multipleSelect--control {
  border: 1px solid;
}

ui-fss.g85[id="c15"] {
  content: "c16,";
}

.c3 {
  height: 1px;
  background: linear-gradient(90deg,var(--navds-semantic-color-canvas-background) 0%,var(--navds-semantic-color-border) 5%,var(--navds-semantic-color-border) 95%,var(--navds-semantic-color-canvas-background) 100% );
  width: -webkit-fill-available;
  margin-left: 2rem;
  margin-right: 2rem;
}

ui-fss.g89[id="c2"] {
  content: "c3,";
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

ui-fss.g100[id="c11"] {
  content: "c12,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g104[id="c0"] {
  content: "c1,";
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="c0 c1"
>
  <h1
    class="navds-heading navds-heading--medium"
  >
    buc:step-startSEDTitle
  </h1>
  <div
    class="c2 c3"
  />
  <div
    class="c4 c5"
  >
    <div
      class="c6 c7"
    >
      <div
        class="c8 c9"
        size="2"
      />
      <label
        class="navds-text-field--label navds-label"
      >
        buc:form-chooseSed
      </label>
      <div
        class=""
        data-testid="a_buc_c_sedstart--sed-select-id"
      >
        <div
          class=" css-91gbow-container"
          id="a_buc_c_sedstart--sed-select-id"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-4-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1q6azai-control"
          >
            <div
              class=" css-319lph-ValueContainer"
            >
              <div
                class=" css-14el2xx-placeholder"
                id="react-select-4-placeholder"
              />
              <div
                class=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-4-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="a_buc_c_sedstart--sed-select-id"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1pm5dtp-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c8 c10"
      />
      <div
        class="c11 c12"
        data-testid="a_buc_c_sedstart--avdod-div-id"
      >
        <svg
          fill="none"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9 7a3 3 0 1 1 6 0 3 3 0 0 1-6 0Zm3-5a5 5 0 1 0 0 10 5 5 0 0 0 0-10ZM6 21a6 6 0 0 1 12 0v1h2v-1a8 8 0 1 0-16 0v1h2v-1Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="c13 c14"
        />
        <label
          class="navds-text-field--label navds-label"
        >
          buc:form-avdod
          :
        </label>
        <div
          class="c13 c14"
        />
        <p
          class="navds-body-long"
        >
          MOR AVDÃ˜D1 - personFarFnr (buc:relasjon-MOR)
        </p>
      </div>
      <div
        class="c8 c10"
      />
      <div
        class=""
      >
        <label
          class="navds-text-field__label navds-label"
          for="a_buc_c_sedstart--country-select-id"
        >
          buc:form-chooseCountry
        </label>
        <div
          class="c-countrySelect__select css-91gbow-container"
        >
          <span
            class="css-7pg0cj-a11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-7pg0cj-a11yText"
          />
          <div
            class="c-countrySelect__select__control css-1eu03fr-control"
          >
            <div
              class="c-countrySelect__select__value-container c-countrySelect__select__value-container--is-multi css-1d8n9bt"
            >
              <div
                class="c-countrySelect__select__placeholder css-14el2xx-placeholder"
                id="react-select-3-placeholder"
              />
              <div
                class="c-countrySelect__select__input-container css-ackcql"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-3-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="ui:country"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="c-countrySelect__select__input"
                  id="a_buc_c_sedstart--country-select-id"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="c-countrySelect__select__indicators css-1wy0on6"
            >
              <span
                class="c-countrySelect__select__indicator-separator css-1pm5dtp-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="c-countrySelect__select__indicator c-countrySelect__select__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-8mmkcg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c8 c10"
      />
      <div
        class="c15 c16"
      >
        <label
          class="navds-text-field--label navds-label"
          for="a_buc_c_sedstart--institution-select-id"
        >
          buc:form-chooseInstitution
        </label>
        <div
          class="multipleSelect css-91gbow-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-5-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="multipleSelect__control css-1q6azai-control"
          >
            <div
              class="multipleSelect__value-container multipleSelect__value-container--is-multi css-319lph-ValueContainer"
            >
              <div
                class="multipleSelect__placeholder css-14el2xx-placeholder"
                id="react-select-5-placeholder"
              />
              <div
                class="multipleSelect__input-container css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-5-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="ui:institution"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="multipleSelect__input"
                  id="a_buc_c_sedstart--institution-select-id"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="multipleSelect__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="multipleSelect__indicator-separator css-141yg44-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="multipleSelect__indicator multipleSelect__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c8 c9"
        size="2"
      />
      <label
        class="navds-text-field--label navds-label"
      >
        buc:form-chosenInstitutions
      </label>
      <div
        class="c8 c10"
      />
      <div
        class=""
      >
        <div
          class="c17 c18"
        >
          <p
            class="navds-body-long"
          >
            buc:form-noInstitutionYet
          </p>
        </div>
      </div>
      <div
        class="c6 c7"
      >
        <div
          class="c8 c19"
          size="1.5"
        />
        <button
          class="navds-button navds-button--primary navds-button--medium"
          data-amplitude="sed.new.create"
          data-testid="a_buc_c_sedstart--forward-button-id"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            buc:form-orderSED
          </span>
        </button>
        <div
          class="c13 c14"
        />
        <button
          class="navds-button navds-button--tertiary navds-button--medium"
          data-amplitude="sed.new.cancel"
          data-testid="a_buc_c_sedstart--cancel-button-id"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            ui:cancel
          </span>
        </button>
        <div
          class="c8 c19"
          size="1.5"
        />
      </div>
    </div>
    <div
      class="c13 c14"
    />
    <div
      class="c6 c7"
    >
      <div
        class="c6 c7"
      />
    </div>
  </div>
</div>
`;
